### PR TITLE
Add some more commands for installing/removing/listing versions

### PR DESCRIPTION
**Pull request by urso**
- Package github.com/anrewkroh/gvm acts as a library now. One can use `gvm.Manager` for writing custom tools.
- Versions are parsed now. E.g. one can use version `1.11beta1` or `1.11-beta1` to download the go 1.11 beta1 release. The version formatter ensures the correct file is found for downloading.
- Moved CLI to github.com/anrewkroh/gvm/cmd/gvm
  - Add support for new commands:
    - use: implements old behavior. Install and write CLI settings to stdout
    - install: download binary version
    - available: print list of installable go releases
    - list: list all installed versions
    - remove: remove some versions
    - purge: remove all but latest stable/unstable version
  - Default command is `use`. CLI can be used as before

### CODE REVIEW

File: cmd/gvm/gvm.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/andrewkroh/gvm"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const usage = `gvm is a Go version manager. gvm installs a Go version and prints
+the commands to configure your environment to use it. gvm can only install
+binary versions of Go from https://golang.org/dl/. Below are examples for
```
Comment (andrewkroh) [inline cmd/gvm/gvm.go]: I guess it's not limited to "binary versions" now.

File: cmd/gvm/internal/shellfmt/format.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+	"github.com/pkg/errors"
+)
+
+type Fmt struct {
+	out io.Writer
+	fmt EnvFormatter
+}
+
+type EnvFormatter interface {
```
Comment (andrewkroh) [inline cmd/gvm/internal/shellfmt/format.go]: Much nicer!

File: gvm.go [hunk 3]
```diff
@@ -1,163 +1,134 @@
...
 	}
 
+	m.cacheDir = filepath.Join(m.Home, "cache")
+	m.versionsDir = filepath.Join(m.Home, "versions")
+	return m.ensureDirStruct()
+}
+
+func (m *Manager) ensureDirStruct() error {
+	for _, dir := range []string{m.cacheDir, m.versionsDir} {
+		if err := os.MkdirAll(dir, os.ModeDir|os.ModePerm); err != nil {
```
Comment (andrewkroh) [inline gvm.go]: `os.ModeDir|os.ModePerm` - That's like 0777, right? That's too liberal for security purposes. I suggest `0755`.

File: cmd/gvm/gvm.go (hunk 1 above)
Comment (urso) [reply cmd/gvm/gvm.go]: It's still limited to binaries for now. Due to the many refactorings I'd rather want to add compile support in another PR.

**Comment by andrewkroh** [comment]
I get an error when running `available` on mac and win. Everything else works as I would expect.

```
$ ./gvm available
gvm: error: Get /Users/akroh/.gvm?marker=: unsupported protocol scheme ""
```

```
PS C:\Gopath\src\github.com\elastic\beats> .\gvm.exe available
gvm: error: Get c:\Users\vagrant\.gvm?marker=: unsupported protocol scheme "c"
```

**Comment by andrewkroh** [comment]
I working on adding AppVeyor BTW. There's something wrong with my account and I can't add or remove any project so it's going to take a bit b/c I need to contact support.

**Pull request merged**